include/inviwo/core/util/sourcecontext.h

```diff
@@ -0,0 +1,142 @@
+/*********************************************************************************
+ *
+ * Inviwo - Interactive Visualization Workshop
+ *
+ * Copyright (c) 2019 Inviwo Foundation
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ * list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ *********************************************************************************/
+#pragma once
+
+#include <inviwo/core/common/inviwocoredefine.h>
+#include <inviwo/core/util/stringconversion.h>
+#include <inviwo/core/util/hashcombine.h>
+
+#include <string>
+#include <iostream>
+
+namespace inviwo {
+
+/**
+ * Represents a location in source code
+ */
+class IVW_CORE_API SourceContext {
```

Comment:
Would be more clear if this file (and class) would be called SourceCodeContext. Perhaps we should even put this in a subfolder (debug) since it is just for debugging?

Reply:
That's not just for debugging. Exception handling, stack traces, etc. Regarding the naming scheme, it was previously dubbed `ExceptionContext`. It refers to where its coming from, i.e. the source/reason, not necessarily source code.

Reply:
It says "a location in source code" and it has file_ and function_ as member variables :)

Reply:
wanted to be constant with https://en.cppreference.com/w/cpp/experimental/source_location

Reply:
So we don't want to name it SourceLocation?

Reply:
We have a SourceLocation further down... that is new and constepr but has a few restrictions... will add some comments...